dask/dataframe/io/hdf.py

```diff
@@ -452,14 +452,14 @@ def _build_parts(paths, key, start, stop, chunksize, sorted_index, mode):
             path, key, stop, sorted_index, chunksize, mode
         )
 
-        for k, stop, division in zip(keys, stops, divisions):
+        for k, s, d in zip(keys, stops, divisions):
 
-            if division and global_divisions:
-                global_divisions = global_divisions[:-1] + division
-            elif division:
-                global_divisions = division
+            if d and global_divisions:
+                global_divisions = global_divisions[:-1] + d
+            elif d:
+                global_divisions = d
 
-            parts.extend(_one_path_one_key(path, k, start, stop, chunksize))
+            parts.extend(_one_path_one_key(path, k, start, s, chunksize))
```

Comment:
Are all these changes just renaming `stop` to `s` and `division` to `d`?

Reply:
Yeah, it looks like we're currently overwriting the `stop` variable which is passed into `_build_parts`, which is causing issues when calling `_get_keys_stops_divisions` in the `for`-loop over `paths`

Reply:
exactly

Reply:
Perfect - That makes sense! Thanks for explaining (hard to see from the diff).

Reply:
Totally. Incidentally I think there are linters that would catch this kind of bad behavior, but that's a whole other can of worms.